x-pack/plugins/observability/public/pages/slo_edit/helpers/process_slo_form_values.ts

```diff
@@ -91,21 +100,50 @@ export function transformValuesToUpdateSLOInput(values: CreateSLOForm): UpdateSL
   };
 }
 
-export function transformPartialCreateSLOInputToPartialCreateSLOForm(
-  values: Partial<CreateSLOInput>
-): Partial<CreateSLOForm> {
-  return {
-    ...values,
-    ...(values.objective && {
-      objective: {
-        target: values.objective.target * 100,
-        ...(values.objective.timesliceTarget && {
-          timesliceTarget: values.objective.timesliceTarget * 100,
-        }),
-        ...(values.objective.timesliceWindow && {
-          timesliceWindow: String(toDuration(values.objective.timesliceWindow).value),
-        }),
-      },
-    }),
-  };
+function transformPartialIndicatorState(
+  indicator?: RecursivePartial<Indicator>
+): Indicator | undefined {
+  if (indicator === undefined || indicator.type === undefined) return undefined;
+
+  const indicatorType = indicator.type;
+  switch (indicatorType) {
+    case 'sli.apm.transactionDuration':
```

Comment:
nit: It would be nice to have constants for the different indicator types. But I know we use these strings in many places, so it would require more changes.

Reply:
Those are typed. If you try to use another string, or add a new type in the indicator type definition but forget to handle it here, you'll get a TS error